lib/core/test/unit/Cardano/Wallet/Primitive/AddressDiscovery/RandomSpec.hs

```diff
@@ -0,0 +1,141 @@
+    describe "Golden tests for Byron Addresses w/ random scheme (Mainnet)" $ do
+    it "check isOurs for initial account" $
+        checkIsOurs DecodeDerivationPath
+        { mnem =
+                arbitraryMnemonic
+        , addr =
+                "82d818584283581ca08bcb9e5e8cd30d5aea6d434c46abd8604fe4907d\
+                \56b9730ca28ce5a101581e581c22e25f2464ec7295b556d86d0ec33bc1\
+                \a681e7656da92dbc0582f5e4001a3abe2aa5"
+        , accIndex =
+                2147483648
+        , addrIndex =
+                2147483648
+        }
+    it "check isOurs for another account" $
+        checkIsOurs DecodeDerivationPath
+        { mnem =
+                arbitraryMnemonic
+        , addr =
+                "82d818584283581cb039e80866203e82fc834b8e6a355b83ec6f8fd199\
+                \66078a40e6d6b2a101581e581c22e27fb12d08728073cd416dfbfcb8dc\
+                \0e760335d1d60f65e8740034001a4bce4d1a"
+        , accIndex =
+                2694138340
+        , addrIndex =
+                2512821145
+        }
+
+goldenSpecTestnet :: Spec
+goldenSpecTestnet =
+    describe "Golden tests forByron Addresses w/ random scheme (Testnet)" $ do
+    it "check isOurs - initial account" $
+        checkIsOurs DecodeDerivationPath
+        { mnem =
+                arbitraryMnemonic
+        , addr =
+                "82d818584983581ca03d42af673855aabcef3059e21c37235ae706072d\
+                \38150dcefae9c6a201581e581c22e25f2464ec7295b556d86d0ec33bc1\
+                \a681e7656da92dbc0582f5e402451a4170cb17001a39a0b7b5"
+        , accIndex =
+                2147483648
+        , addrIndex =
+                2147483648
+        }
+    it "check isOurs - another account" $
+        checkIsOurs DecodeDerivationPath
+        { mnem =
+                arbitraryMnemonic
+        , addr =
+                "82d818584983581c267b40902921c3afd73926a83a23ca08ae9626a64a\
+                \4b5616d14d6709a201581e581c22e219c90fb572d565134f6daeab650d\
+                \c871d130430afe594116f1ae02451a4170cb17001aee75f28a"
+        , accIndex =
+                3337448281
+        , addrIndex =
+                3234874775
+        }
+
+{-------------------------------------------------------------------------------
+                    Golden tests for Address derivation path
+-------------------------------------------------------------------------------}
+
+data DecodeDerivationPath = DecodeDerivationPath
+    { mnem :: [Text]
+    , addr :: ByteString
+    , accIndex :: Word32
+    , addrIndex :: Word32
+    } deriving (Show, Eq)
+
+-- An aribtrary mnemonic sentence for the tests
+arbitraryMnemonic :: [Text]
+arbitraryMnemonic =
+    [ "price", "whip", "bottom", "execute", "resist", "library"
+    , "entire", "purse", "assist", "clock", "still", "noble" ]
+
+
+checkIsOurs :: DecodeDerivationPath -> Expectation
+checkIsOurs DecodeDerivationPath{..} =
+    isOurs (Address addr) (RndState rootXPrv) `shouldBe` (True, RndState rootXPrv)
```

Comment:
Would be good to also have negative tests here? What about putting the expectation in the test object (and maybe rename it as `DecodeDerivationPath` seems like a copy/pasting from elsewhere :p), such that we can easily define positive and negative tests.

Reply:
I added negative tests for a bogus address, and wrong seed.